main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala

```diff
@@ -870,4 +925,129 @@ class TestTyper extends FunSuite with TestUtils {
     expectError[TypeError.MismatchedBools](result)
   }
 
+  test("Test.OverlappingInstance.01") {
+    val input =
+      """
+        |class C[a]
+        |
+        |instance C[Int]
+        |
+        |instance C[Int]
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[TypeError.OverlappingInstance](result)
+  }
+
+
+  test("Test.OverlappingInstance.02") {
+    val input =
+      """
+        |class C[a]
+        |
+        |instance C[a]
+        |
+        |instance C[Int]
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[TypeError.OverlappingInstance](result)
+  }
+
+  test("Test.OverlappingInstance.03") {
+    val input =
+      """
+        |class C[a]
+        |
+        |instance C[(a, Int)]
+        |
+        |instance C[(Bool, b)]
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[TypeError.OverlappingInstance](result)
+  }
```

Comment:
With the current implementation:
* Any two schema instances are overlapping (since Schemas are implicitly open, right? Otherwise the case is the same
* Any two polymorphic records are overlapping, since they can always unify.